software/bootcamp-code/main.py

```diff
@@ -33,56 +56,170 @@ def main_loop():
-        # Ball desired coordinates are in the middle of the frame.
-        ball_desired_x = cam.rgb_width/2
-        ball_desired_y = cam.rgb_height/2
+    
+    # time counter for testing
+    # zero_time = time.time()
         
+    # Ball desired coordinates are in the middle of the frame width and 83% of frame height
+    # (THE ZERO COORDINATES OF THE FRAME ARE IN THE UPPER LEFT CORNER.)
+    ball_desired_x = cam.rgb_width/2
+    ball_desired_y = cam.rgb_height/1.2
+
+    # Speed range for motors is 48 - 2047, we use 100 for max motor speed at the moment.
+    max_motor_speed = 80
+    # rotation speed for find ball state
+    find_rotation_speed = max_motor_speed/6
+    # variable to store target basket color, currently blue for testing
+    basketColor = BasketColor.BLUE
+    # orbiting speed for centering the basket
+    orbit_speed = max_motor_speed/1.5
+    # start time variable for thrower state
+    throw_start=0
+    # true false variable to keep track if when throwing, the ball has left the camera frame
+    ball_out_of_frame=False
+    # when throwing the ball, the speed which the robot moves forward
+    throw_forward_speed=max_motor_speed/4
+    # the maximum distance of the ball (y zero coordinate is in the upper edge of the frame)
+    basket_max_distance=cam.rgb_height
+    # maximum speed to throw the ball
+    throw_motor_speed_max=max_motor_speed/8
+
+    # for printing logs (log when change state)
+    new_state=True
+    last_state=State.FIND_BALL
+    
+    try:        
         while True:
+            if new_state==True: print(state)
             # has argument aligned_depth that enables depth frame to color frame alignment. Costs performance
             processedData = processor.process_frame(aligned_depth=False)
-
-            # TODO This is where you add the driving behaviour of your robot. It should be able to filter out
-            # objects of interest and calculate the required motion for reaching the object.
-
-            # STATE TO FIND THE BALL 0
+            
+            # Here is the state machine for the driving logic.
+                #Omni-motion movement:
+                # side speed is x_speed
+                # forward speed is y_speed
+                # rotation if you want to turn
+            
+            # STATE TO FIND THE BALL
             if state == State.FIND_BALL:
+                last_state, new_state =  state_printer(state, last_state, new_state)
+
+                # if we have a ball in view, center it
                 if len(processedData.balls)>0:
-                    state=State.CENTER_BALL
+                    state=State.MOVE_CENTER_BALL
+                    continue
+                # if no ball found, rotate
+                
+                omni_motion.move(0, 0, find_rotation_speed)
+
+            # STATE TO MOVE TO AND CENTER THE BALL
+            elif state == State.MOVE_CENTER_BALL:
+                last_state, new_state =  state_printer(state, last_state, new_state)
```

Comment:
IS this method used for every state? Is it required to be in every state separately then?